### ISSUE TITLE
[v5.3.0] InflateException - AppCompatAutoCompleteTextView

### ISSUE THREAD

**Issue by acouhier-luna** [Adyen/adyen-android#1523]
**Describe the bug**
When I want to pay with a new credit card, I crash

**To Reproduce**
1. Click on `Credit Card`

**Expected behavior**
I would like to use a new credit card

**Screenshots**
| error | what I expect |
|--|--|
| Crash when clicking on `Credit card` ![image](https://github.com/Adyen/adyen-android/assets/102948122/1177bb61-8b4d-400a-b584-73391690464d) | <img width="362" alt="image" src="https://github.com/Adyen/adyen-android/assets/102948122/9f33d4b8-9a86-4711-bc2e-7894d92794bd"> |

**Smartphone (please complete the following information):**
 - Device: Galaxy A10
 - OS:Android version : 11
 - Browser : Chrome

**Additional context**
<details>
<summary>Logcat</summary>

```
FATAL EXCEPTION: main
                                         Process: XXX, PID: 2890
                                         android.view.InflateException: Binary XML file line #166 in XXX:layout/card_view: Binary XML file line #166 in XXX:layout/card_view: Error inflating class androidx.appcompat.widget.AppCompatAutoCompleteTextView
                                         	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@e5b11a3, Dispatchers.Main.immediate]
                                         Caused by: android.view.InflateException: Binary XML file line #166 in XXX:layout/card_view: Error inflating class androidx.appcompat.widget.AppCompatAutoCompleteTextView
                                         Caused by: java.lang.reflect.InvocationTargetException
                                         	at java.lang.reflect.Constructor.newInstance0(Native Method)
                                         	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
                                         	at android.view.LayoutInflater.createView(LayoutInflater.java:852)
                                         	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1004)
                                         	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:959)
                                         	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1121)
                                         	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1082)
                                         	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1124)
                                         	at android.view.LayoutInflater.inflate(LayoutInflater.java:654)
                                         	at android.view.LayoutInflater.inflate(LayoutInflater.java:532)
                                         	at android.view.LayoutInflater.inflate(LayoutInflater.java:479)
                                         	at com.adyen.checkout.card.databinding.CardViewBinding.inflate(CardViewBinding.java:185)
                                         	at com.adyen.checkout.card.internal.ui.view.CardView.<init>(CardView.kt:74)
                                         	at com.adyen.checkout.card.internal.ui.view.CardView.<init>(CardView.kt:62)
                                         	at com.adyen.checkout.card.internal.ui.CardViewProvider.getView(CardViewProvider.kt:28)
                                         	at com.adyen.checkout.ui.core.AdyenComponentView.loadView(AdyenComponentView.kt:121)
                                         	at com.adyen.checkout.ui.core.AdyenComponentView.access$loadView(AdyenComponentView.kt:48)
                                         	at com.adyen.checkout.ui.core.AdyenComponentView$attach$1.invokeSuspend(AdyenComponentView.kt:104)
                                         	at com.adyen.checkout.ui.core.AdyenComponentView$attach$1.invoke(Unknown Source:8)
                                         	at com.adyen.checkout.ui.core.AdyenComponentView$attach$1.invoke(Unknown Source:4)
                                         	at kotlinx.coroutines.flow.FlowKt__TransformKt$onEach$$inlined$unsafeTransform$1$2.emit(Emitters.kt:219)
                                         	at kotlinx.coroutines.flow.StateFlowImpl.collect(StateFlow.kt:392)
                                         	at kotlinx.coroutines.flow.StateFlowImpl$collect$1.invokeSuspend(Unknown Source:15)
                                         	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                         	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
                                         	at kotlinx.coroutines.EventLoop.processUnconfinedEvent(EventLoop.common.kt:65)
                                         	at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:371)
                                         	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:26)
                                         	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:21)
                                         	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:88)
                                         	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:123)
                                         	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:52)
                                         	at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
                                         	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:43)
                                         	at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
                                         	at kotlinx.coroutines.flow.FlowKt__CollectKt.launchIn(Collect.kt:45)
                                         	at kotlinx.coroutines.flow.FlowKt.launchIn(Unknown Source:1)
                                         	at com.adyen.checkout.ui.core.AdyenComponentView.attach(AdyenComponentView.kt:111)
                                         	at com.adyen.checkout.dropin.internal.ui.CardComponentDialogFragment.onViewCreated(CardComponentDialogFragment.kt:52)
                                         	at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3147)
                                         	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:588)
10:40:46.835 AndroidRuntime           E  	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:272)
                                         	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1943)
                                         	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1845)
                                         	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1782)
                                         	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:565)
                                         	at android.os.Handler.handleCallback(Handler.java:938)
                                         	at android.os.Handler.dispatchMessage(Handler.java:99)
                                         	at android.os.Looper.loop(Looper.java:246)
                                         	at android.app.ActivityThread.main(ActivityThread.java:8625)
                                         	at java.lang.reflect.Method.invoke(Native Method)
                                         	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
                                         	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
                                         Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 11: TypedValue{t=0x2/d=0x1010098 a=6}
                                         	at android.content.res.TypedArray.getColorStateList(TypedArray.java:597)
                                         	at androidx.appcompat.widget.TintTypedArray.getColorStateList(TintTypedArray.java:185)
                                         	at androidx.appcompat.widget.AppCompatTextHelper.loadFromAttributes(AppCompatTextHelper.java:315)
                                         	at androidx.appcompat.widget.AppCompatAutoCompleteTextView.<init>(AppCompatAutoCompleteTextView.java:99)
                                         	at androidx.appcompat.widget.AppCompatAutoCompleteTextView.<init>(AppCompatAutoCompleteTextView.java:79)
                                         	... 53 more
```
</details>

**Comment by qmetzler-luna** [comment]
Crash is in version v5.3.0 but not in v5.2.0

**Comment by markosopcic** [comment]
Hey! I've got the same issue like  @acouhier-luna, down to the line in the xml where it crashes. This is my first time integrating this SDK on version 5.3.0 so can't say i had it working on previous verisons.  

I tried updating app dependencies to the latest values in the release notes you mentioned, but it didn't help.

Setting the style to the default works:
`<style name="AdyenCheckout" parent="Adyen" />`

If, for example, i set the color primary in the theme to an attribute of my app theme, e.g.:

`    <style name="AdyenCheckout" parent="Adyen">
        <item name="colorPrimary">?attr/idBasePrimary</item>
    </style>`
I get another exception thrown:
<details>
<summary>Logcat</summary>

```
2024-03-29 14:01:23.850 10455-10455 AndroidRuntime          com.my.app.package.name     E  FATAL EXCEPTION: main
                                                                                                    Process: com.my.app.package.name, PID: 10455
                                                                                                    android.view.InflateException: Binary XML file line #32 in com.my.app.package.name:layout/payment_methods_list_header: Binary XML file line #32 in com.my.app.package.name:layout/payment_methods_list_header: Error inflating class TextView
                                                                                                    Caused by: android.view.InflateException: Binary XML file line #32 in com.my.app.package.name:layout/payment_methods_list_header: Error inflating class TextView
                                                                                                    Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 5: TypedValue{t=0x2/d=0x7f0401ae a=8}, theme={InheritanceMap=[id=0x7f150031com.my.app.package.name:style/AdyenCheckout.BottomSheetDialogTheme, id=0x7f1504fdcom.my.app.package.name:style/ThemeOverlay.MaterialComponents.BottomSheetDialog, id=0x7f1501b6com.my.app.package.name:style/Base.V21.ThemeOverlay.MaterialComponents.BottomSheetDialog, id=0x7f1501a6com.my.app.package.name:style/Base.V14.ThemeOverlay.MaterialComponents.BottomSheetDialog, id=0x7f15018ecom.my.app.package.name:style/Base.ThemeOverlay.MaterialComponents.Dialog, id=0x7f1501a7com.my.app.package.name:style/Base.V14.ThemeOverlay.MaterialComponents.Dialog, id=0x7f1504aecom.my.app.package.name:style/ThemeOverlay.AppCompat.Dialog, id=0x7f150186com.my.app.package.name:style/Base.ThemeOverlay.AppCompat.Dialog, id=0x7f1501b3com.my.app.package.name:style/Base.V21.ThemeOverlay.AppCompat.Dialog, id=0x7f1501c8com.my.app.package.name:style/Base.V7.ThemeOverlay.AppCompat.Dialog, id=0x7f150182com.my.app.package.name:style/Base.ThemeOverlay.AppCompat, id=0x7f15029ccom.my.app.package.name:style/Platform.ThemeOverlay.AppCompat], Themes=[com.my.app.package.name:style/AdyenCheckout.BottomSheetDialogTheme, forced, com.my.app.package.name:style/AdyenCheckout.Translucent, forced, com.my.app.package.name:style/Theme.AppCompat.Empty, forced, android:style/Theme.DeviceDefault.Light.DarkActionBar, forced]}
                                                                                                    	at android.content.res.TypedArray.getColorStateList(TypedArray.java:600)
                                                                                                    	at android.widget.TextView.readTextAppearance(TextView.java:4251)
                                                                                                    	at android.widget.TextView.<init>(TextView.java:1249)
                                                                                                    	at android.widget.TextView.<init>(TextView.java:1149)
                                                                                                    	at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:113)
                                                                                                    	at com.google.android.material.textview.MaterialTextView.<init>(MaterialTextView.java:92)
                                                                                                    	at com.google.android.material.textview.MaterialTextView.<init>(MaterialTextView.java:87)
                                                                                                    	at com.google.android.material.theme.MaterialComponentsViewInflater.createTextView(MaterialComponentsViewInflater.java:61)
                                                                                                    	at androidx.appcompat.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:148)
                                                                                                    	at androidx.appcompat.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1678)
                                                                                                    	at androidx.appcompat.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1729)
                                                                                                    	at android.view.LayoutInflater.tryCreateView(LayoutInflater.java:1073)
                                                                                                    	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1009)
                                                                                                    	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:973)
                                                                                                    	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1135)
                                                                                                    	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1096)
                                                                                                    	at android.view.LayoutInflater.inflate(LayoutInflater.java:694)
                                                                                                    	at android.view.LayoutInflater.inflate(LayoutInflater.java:538)
                                                                                                    	at com.adyen.checkout.dropin.databinding.PaymentMethodsListHeaderBinding.inflate(PaymentMethodsListHeaderBinding.java:49)
                                                                                                    	at com.adyen.checkout.dropin.internal.ui.PaymentMethodAdapter.onCreateViewHolder(PaymentMethodAdapter.kt:53)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:7788)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6873)
2024-03-29 14:01:23.851 10455-10455 AndroidRuntime          com.my.app.package.name     E  	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6757)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6753)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2362)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1662)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4645)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:4022)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
                                                                                                    	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
                                                                                                    	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at com.google.android.material.bottomsheet.BottomSheetBehavior.onMeasureChild(BottomSheetBehavior.java:515)
                                                                                                    	at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:831)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                                                    	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:145)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
                                                                                                    	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
                                                                                                    	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7008)
                                                                                                    	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                                                    	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:750)
                                                                                                    	at android.view.View.measure(View.java:27122)
                                                                                                    	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:4182)
                                                                                                    	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:2759)
                                                                                                    	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3086)
                                                                                                    	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2465)
                                                                                                    	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9305)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1339)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1348)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:952)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:882)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1322)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205)
                                                                                                    	at android.os.Looper.loop(Looper.java:294)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8177)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
2024-03-29 14:01:23.851 10455-10455 AndroidRuntime          com.my.app.package.name     E  	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```
</details>

While if i reference a color directly everything seems to work. 
Please let me know if i can provide any more information.

**Comment by markosopcic** [comment]
Also, for some reason, overriding the following, without overriding the `AdyenCheckout` style also seems to work, don't know why.

```
    <style name="AdyenCheckout.Card.AddressLookup.DropdownTextInputEditText" parent="AdyenCheckout.DropdownTextInputEditText">
        <item name="android:layout_height">wrap_content</item>
        <item name="android:layout_width">match_parent</item>
    </style>
    ```

**Comment by jreij** [comment]
@markosopcic thanks for the detailed info! We'll investigate this and get back to you when we have an update.

**Comment by jreij** [comment]
@acouhier-luna @qmetzler-luna @markosopcic
Hi again, could you try the following solutions and let us know which one(s) solve the issue:
- Set up the SDK with the default theming by adding this line in your `styles.xml` file: `<style name="AdyenCheckout" parent="Adyen" />`.
- Set up the SDK with your own theme by extending from your own app theme and define the `android:textColor` attribute there (or directly in your app theme):
```xml
    <style name="AdyenCheckout" parent="YOUR_APP_THEME">
        <item name="android:textColor">@color/text_color_primary</item>
        ...
    </style>
```
You might need to define other attributes, take a look at [MyAppTheme](https://github.com/Adyen/adyen-android/blob/main/example-app/src/main/res/values/styles.xml#L4) in our example app for guidance.

**Comment by markosopcic** [comment]
Hey @jreij!
- When setting the SDK up with the default theming, everything works, but as expected, it uses the default theme from the app.
- when extending the `AdyenCheckout` theme from my app theme, and setting the `android:textColor`, property, it works as well, but the colors from the app theme don't get recognized, even though we have them set. This is what the drop in component looks like in that case (i barely even noticed the cancel button on the "awaiting completion screen"):
<img width="268" alt="Screenshot 2024-04-05 at 13 46 54" src="https://github.com/Adyen/adyen-android/assets/32719316/b9ebe422-2324-4b3d-8064-b523cea09a40">
<img width="268" alt="Screenshot 2024-04-05 at 13 47 09" src="https://github.com/Adyen/adyen-android/assets/32719316/57d1eb90-8bed-471e-92b5-067d4ee95bcf">

- after i set most of the values to the same color (#001B40) that are set in the `MyAppTheme` example, i.e.:

```
    <style name="AdyenCheckout" parent="AppTheme" >
        <item name="android:textColor">@color/id_blue_100</item>
        <item name="colorPrimary">@color/id_blue_100</item>
        <item name="colorOnPrimary">@color/id_blue_100</item>
        <item name="colorSecondary">@color/id_blue_100</item>
        <item name="colorOnSecondary">@color/id_blue_100</item>
        <item name="colorAccent">@color/id_blue_100</item>
        <item name="android:colorBackground">@color/id_blue_100</item>
        <item name="colorOnBackground">@color/id_blue_100</item>
        <item name="colorSurface">@color/id_blue_100</item>
        <item name="colorOnSurface">@color/id_blue_100</item>
        <item name="colorControlNormal">@color/id_blue_100</item>
        <item name="colorControlHighlight">@color/id_blue_100</item>

        <item name="android:textColorPrimary">@color/id_blue_100</item>
        <item name="android:textColorSecondary">@color/id_blue_100</item>
        <item name="android:textColorTertiary">@color/id_blue_100</item>
        <item name="android:textColorLink">@color/id_blue_100</item>
    </style>
```
It doesn't look like the color is applied anywhere, i.e.:
<img width="252" alt="image" src="https://github.com/Adyen/adyen-android/assets/32719316/4fc3065c-cf66-4313-88b0-fc6af3318811">
<img width="252" alt="image" src="https://github.com/Adyen/adyen-android/assets/32719316/3699fae2-62c5-42ce-9c70-fa087c7b36ec">

**Comment by jreij** [comment]
Hi @markosopcic , thanks for the detailed info!
We just opened a PR that should remove the need to manually add `android:textColor` to the styles, it will be included in the next release and you can remove this temp fix after upgrading.

As for the rest of the styling, I just tried your styling code with this color `<color name="id_blue_100">#001B40</color>` and my whole drop-in is blue 😄 
Could you maybe share a few more things:
- Is your main theme called `AppTheme`? You can find that in your manifest file under `application` -> `android:theme`.
- Which theme does your `AppTheme` extend from? Feel free to share the whole xml block where you define it.
- Do you support dark mode? And if so, are the results the same in light and dark mode?

**Comment by OscarSpruit** [comment]
Hi, we just released [5.3.1](https://github.com/Adyen/adyen-android/releases/tag/5.3.1) in which this issue should be fixed.
